### PR TITLE
Migrate test_withExperiment.js

### LINKED ISSUE #11275: Migrate /unit/amo/test_withExperiment.js to react-testing-library
Migrate /unit/amo/test_withExperiment.js to react-testing-library

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #11275 

I decided to tackle this one as I was working on TestAddon, which led me to TestPage, which led me to TestVPNPromoBanner, which eventually led me to this component. The changes here are pretty straightforward, but I faced some challenges with dealing with cookies in other components, which I will tackle next.

### CODE REVIEW

File: tests/unit/helpers.js [hunk 1]
```diff
@@ -1312,6 +1312,17 @@ export function fakeCookies(overrides = {}) {
   };
 }
 
+export function fakeCookiesWithJest(overrides = {}) {
```
Comment (bobsilverberg) [inline tests/unit/helpers.js:1315]: `fakeCookies` is used by other tests that have not been converted yet, so I created this version for the updated tests. Once everything that uses this has been converted we can delete `fakeCookies` and rename `fakeCookiesWithJest` to `fakeCookies`.

File: tests/unit/helpers.js [hunk 2]
```diff
@@ -1322,6 +1333,16 @@ export const createFakeTracking = (overrides = {}) => {
   };
 };
 
+export const createFakeTrackingWithJest = (overrides = {}) => {
```
Comment (bobsilverberg) [inline tests/unit/helpers.js:1336]: `createFakeTracking` is used by other tests that have not been converted yet, so I created this version for the updated tests. Once everything that uses this has been converted we can delete `createFakeTracking` and rename `createFakeTrackingWithJest` to `createFakeTracking`.

**Review by eviljeff — APPROVED** [review]

File: tests/unit/helpers.js (hunk 1 above)
Comment (bobsilverberg) [reply tests/unit/helpers.js:1315]: I did consider it, and didn't do it for exactly the reason you state: trying to keep the current diff smaller. Ultimately it's probably the same amount of effort, but I figured a final PR to just rename everything would be easier to review.

File: tests/unit/helpers.js (hunk 2 above)
Comment (bobsilverberg) [reply tests/unit/helpers.js:1336]: Same answer :-)

**Pull request merged**